src/pages/signup-affiliates/components/_dialog-model.tsx

```diff
@@ -0,0 +1,146 @@
+import React, { Ref, useEffect } from 'react'
+import styled, { css } from 'styled-components'
+import { Flex } from 'components/containers'
+import { Header } from 'components/elements'
+import { Localize } from 'components/localization'
+import device from 'themes/device'
+
+type DialogModelBoxProps = {
+    aria_label?: string
+    closeModal?: () => void
+    is_open: boolean
+    ref?: Ref<HTMLAnchorElement>
+    rel?: string
+    target?: string
+    to?: string
+    toggle: (event: React.MouseEvent<HTMLElement>) => void
+}
+
+const ModalWrapper = styled.div`
+    position: fixed;
+    top: 50%;
+    left: 50%;
+    transform: translate(-50%, -50%);
+    width: 100%;
+    height: 100%;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    z-index: 200;
+`
+
+const ModalCard = styled.div`
+    position: relative;
+    z-index: 210;
+    display: flex;
+    flex-direction: column;
+    border-radius: 6px;
+    box-shadow: 2px 2px 10px rgba(0, 0, 0, 0.3);
+    max-width: 44rem;
+    background: var(--color-white);
+    padding: 2.4rem;
+    width: 440px;
+    height: 180px;
+    @media ${device.mobileL} {
+        width: 80%;
+    }
+`
+
+const Background = styled.div`
+    position: absolute;
+    width: 100%;
+    height: 100%;
+    top: 0;
+    left: 0;
+    background-color: var(--color-black);
+    opacity: 0.4;
+`
+
+const Action = styled(Flex)`
+    justify-content: space-between;
+    align-items: center;
+`
+
+const shared_css = css`
+    border-radius: 6px;
+    padding: 1rem 1.6rem;
+    transition: all 0.25s;
+    height: fit-content;
+    cursor: pointer;
+    text-decoration: none;
+`
+
+const Proceed = styled.a`
+    ${shared_css}
+    border: 2px solid var(--color-grey-5);
+    color: var(--color-black);
+    background: transparent;
+    margin-right: 0.8rem;
+
+    &:hover {
+        background-color: rgba(0, 0, 0, 0.08);
+    }
+`
+
+const Cancel = styled.span`
+    ${shared_css}
+    border: 2px solid var(--color-red);
+    color: var(--color-white);
+    background: var(--color-red);
+
+    &:hover {
+        background-color: var(--color-red-3);
+        border-color: var(--color-red-3);
+    }
+`
+
+const DialogModelBox = ({ toggle, is_open, closeModal, to = '' }: DialogModelBoxProps) => {
```

Comment:
Nitpick: Before reading the text below not clear for what this component created. Would to offer rename component on `ClosingModalWindox(or Box)`